### DISCUSSION TITLE
Issue with nRF-DTM when using Coded PHY

### DISCUSSION THREAD

**Discussion by macfly2000** [olleheugene/nRF-DTM#5] [General]
[Issues.pdf](https://github.com/olleheugene/nRF-DTM/files/11979412/Issues.pdf)

Hello,
I'm trying to use nRF-DTM between 2 DK nRF52840. I modified the firmware to have access to RSSI, things work well with 1Mbs or 2Mbs PHY. However, when I start to use Coded PHY S2 or S8, The issues are described in attached file. It's a very useful tool when trying to evaluate performances of different BLE modules.
Am I facing real bug and if so, do you see a way to solve them?
Thanks in advance for your time and support.

**Comment by olleheugene** [comment]
Hello,
I am so sorry.... I've just seen your comment.
It seems to bug in the tool. I will try to check the issue from tomorrow.
Sorry again too late answer and thanks for the issue details.

**Comment by olleheugene** [comment]
Hello,
I've checked your problem in both firmware and the nRF DTM tool and found the root cause.
The problem is I cannot set my own RSSI command with coded phy because Bluetooth core spec doesn't allow using the vendor-specific command with coded phy as below figure.
![image](https://github.com/olleheugene/nRF-DTM/assets/8719578/24494c4e-3897-405e-876c-1f58239c942c)
Even if I send the RSSI command over the vendor-specific command, the firmware will replace the type to 0xFF to fill the payload 0b11111111.
So, you need to change the firmware code allowing you to use vendor-specific packet type instead of coded phy type but, this is a violation of Bluetooth core specification.
I think you can modify the firmware source code just for testing RSSI of Coded phy. but I don't recommend you use the same firmware for certification.

Try with the below source code.

_uint32_t dtm_cmd(uint16_t cmd)
{
...
#if defined(READING_AVG_RSSI)
    if ((m_radio_mode == RADIO_MODE_MODE_Ble_1Mbit \
        || m_radio_mode == RADIO_MODE_MODE_Ble_2Mbit \
        || m_radio_mode == RADIO_MODE_MODE_Ble_LR125Kbit \
        || m_radio_mode == RADIO_MODE_MODE_Ble_LR500Kbit \
        ) && payload == DTM_PKT_VENDORSPECIFIC)
#else
    if ((m_radio_mode == RADIO_MODE_MODE_Ble_1Mbit || m_radio_mode == RADIO_MODE_MODE_Ble_2Mbit) && payload == DTM_PKT_VENDORSPECIFIC)
#endif
...
}_

Lastly,
I found a bug in the nRF DTM tool when I saw the problem you mentioned.
Thanks a lot! 👍🏼

**Reply by macfly2000** [reply]
Thank you very much Eugene for your time!
